framer/Extras/Preloader.coffee

```diff
@@ -5,88 +5,114 @@ class Preloader extends BaseClass
 
 	constructor: (options={}) ->
 
-		@_images = []
-		@_imagesLoaded = []
+		@_media = []
+		@_mediaLoaded = []
 		@_isLoading = false
-		
+
 		@timeout = 5
 
-		@context = new Context({name: "Preloader"})
+		@context = new Context({parent: Framer.CurrentContext, name: "Preloader"})
```

Comment:
To draw only draw onto the context if there is one. This is nicer if you use the preloader in Studio, it will make the preloader appear on the device screen instead of flashing the device.